src/util/vfs/vfs-mem.c

```diff
@@ -236,8 +237,12 @@ ssize_t _vfmRead(struct VFile* vf, void* buffer, size_t size) {
 ssize_t _vfmWrite(struct VFile* vf, const void* buffer, size_t size) {
 	struct VFileMem* vfm = (struct VFileMem*) vf;
 
-	if (size + vfm->offset >= vfm->size) {
-		size = vfm->size - vfm->offset;
+	if (size + vfm->offset > vfm->size) {
+		vfm->size = size + vfm->offset;
+		if (vfm->size > vfm->bufferSize) {
+			vfm->size = vfm->bufferSize;
+			size = vfm->size - vfm->offset;
+		}
```

Comment:
I think this wound up in the wrong PR, or at least the wrong commit. Could you fixup the history at least?

Reply:
It's `have VFileFromMemory write "expand" size until the buffer size`. Is this better as separate commits or separate PRs?